plugins/runner-ts/src/index.ts

```diff
@@ -95,47 +100,49 @@ export default defineRunner(tsRunnerOptions, async ctx => {
     getCurrentDirectory: () => process.cwd(),
     getDefaultLibFileName: options => ts.getDefaultLibFilePath(options),
     fileExists: fileName => {
-      if (filesCache[fileName]) {
+      if (filesCache[denormalizePathForTSHost(fileName)]) {
         return true;
       }
       return ts.sys.fileExists(fileName);
     },
     getSourceFile(fileName) {
-      if (fileName.endsWith('.d.ts')) {
-        const stats = ctx.fs.statSync(fileName);
-        let file = filesCache[fileName];
+      const denormalziedFileName = denormalizePathForTSHost(fileName);
```

Comment:
After debugging, in a test environment and production I've found that in production, there is no bug with respect to these changes in the TS host.   

In the test environment, fs is used directly which seems to be why the tests fail, in the production environment union-fs is used and it will somehow use fs behind the scenes, and successfully find the file on Windows even if it has a Unix path (also, the `registerInputFile` method initializes the cache with Unix files in production as well, ensuring correct cache hits).  

Also, I manually tested while debugging and it seems that union-fs works with my changes in which we pass a Windows path.

So in the end, it seems like `union-fs` makes normalization of these paths for the host unnecessary, (at least in the cases I've debugged so far).

So in the end we are with a problem of failing tests on Windows.  It seems that we have 2 options, either 1) normalize the paths like I am doing here or 2) use `union-fs` in our unit tests for the ts-runner.

@beshanoe What are your thoughts?